packages/ui/src/views/repo/components/commits-list.tsx

```diff
@@ -52,62 +52,56 @@ export const CommitsList: FC<CommitProps> = ({ data, toCommitDetails, toCode, cl
-                      >
-                        <StackedList.Field
-                          title={
-                            <div className="flex w-full max-w-full flex-col gap-y-1.5">
-                              {toCommitDetails ? (
-                                <p>
-                                  <Link
-                                    className="text-sm font-medium leading-snug hover:underline"
-                                    to={`${toCommitDetails?.({ sha: commit?.sha || '' })}`}
-                                  >
-                                    <span className="truncate">{commit.title}</span>
-                                  </Link>
-                                </p>
-                              ) : (
-                                <span className="truncate text-sm font-medium leading-snug">{commit.title}</span>
-                              )}
-                              <div className="flex items-center gap-x-1.5">
-                                {authorName && <Avatar name={authorName} src={avatarUrl} size="sm" rounded />}
-                                <span className="text-cn-foreground-3">{authorName || ''}</span>
-                                <span className="text-cn-foreground-2">
-                                  committed on {timeAgo(when, { dateStyle: 'medium' })}
-                                </span>
-                              </div>
+                      <Layout.Horizontal className="w-full truncate">
+                        <Link
+                          className="grow overflow-hidden"
+                          onClick={e => {
+                            e.stopPropagation()
+                          }}
+                          key={commit?.sha}
+                          to={`${toCommitDetails?.({ sha: commit?.sha || '' })}`}
+                        >
+                          <div className="flex w-full max-w-full flex-col gap-y-1.5">
+                            {toCommitDetails ? (
+                              <p>
+                                <Link
+                                  className="flex overflow-hidden text-sm font-medium leading-snug hover:underline"
+                                  to={`${toCommitDetails?.({ sha: commit?.sha || '' })}`}
+                                >
+                                  <Text variant="heading-base" color="foreground-1" truncate title={commit.title}>
+                                    {commit.title}
+                                  </Text>
+                                </Link>
+                              </p>
+                            ) : (
+                              <span className="truncate text-sm font-medium leading-snug">{commit.title}</span>
+                            )}
+                            <div className="flex items-center gap-x-1.5">
+                              {authorName && <Avatar name={authorName} src={avatarUrl} size="sm" rounded />}
```

Comment:
can we merge the condition and render them together?

```suggestion
                              {authorName && 
                              <>
                                <Avatar name={authorName} src={avatarUrl} size="sm" rounded />
                                <span className="text-cn-foreground-3">{authorName}</span>
                                </>
                               }
```